hackathon_site/hardware/views.py

```diff
@@ -31,9 +35,38 @@ def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
 
 
-class OrderListView(mixins.ListModelMixin, generics.GenericAPIView):
+class OrderViewSet(
```

Comment:
had to change this to a viewset to specify different serializer for each method

Reply:
Shouldn't need to, you can define `get_serializer_class` on a generic API view as well. You just key your dictionary on `self.request.method`, which will be `GET`, `POST`, etc. I think all the view set does here is let you call the method `create` instead of `post` which would call `self.create`, defined by CreateModelMixin.

I think its fine to convert this back to a generic view, and remove the router from the urls file.